app/models/mixins/cinder_manager_mixin.rb

```diff
@@ -0,0 +1,28 @@
+module CinderManagerMixin
+  extend ActiveSupport::Concern
+
+  included do
+    # TODO: how about many storage managers???
+    # Should use has_many :storage_managers,
+    has_one  :cinder_manager,
+             :foreign_key => :parent_ems_id,
+             :class_name  => "ManageIQ::Providers::StorageManager::CinderManager",
+             :autosave    => true,
+             :dependent   => :destroy
+
+    delegate :cloud_volumes,
+             :cloud_volume_snapshots,
+             :cloud_volume_backups,
+             :to        => :cinder_manager,
+             :allow_nil => true
+
+    private
+
+    def ensure_storage_managers
```

Comment:
So just a note.  When I add the Swift version of this file, I will have to do one of two things - 1) rename this to "ensure_cinder_managers" along with the call to this from openstack/cloud_manager.rb, so I can add a second call to "ensure_swift_managers" appropriately, or 2) add a *top* level "storage_manager_mixin.rb" (yeah, I know we just removed that) - that defines "ensure_storage_managers", which in turn calls ensure_cinder_managers (note the trailing "s") and ensure_swift_managers, and any other cloud storage managers which we have to add later on for other cloud providers.